src/pybind/mgr/dashboard/frontend/src/app/shared/constants/app.constants.ts

```diff
@@ -0,0 +1,6 @@
+export class AppConstants {
+  public static readonly organization = 'ceph';
+  public static readonly projectName = 'Ceph Dashboard';
+  public static readonly copyRight = 'Copyright(c) 2018 Ceph contributors.';
```

Comment:
A hard coded year is problematic because we have to update this every year. I know it's hard to implement, but something that will replace a placeholder with the current year during compile time would be great.

Reply:
ack

Reply:
@votdev the copyright year has to be the one of the [first publication of a work](http://www.contentious.com/2007/01/07/copyright-notice-is-the-year-really-necessary/). If this dashboard was originally started in 2018, no need to update that date then.

Reply:
@epuertat @a2batic 

Regarding copyright year, the best practice is to have the copyright be as current as possible and not have it go back to the first publication of the work.  E.g. If a project is first published/mentioned in 1993, then the time that a copyright is enforceable is shortened (20 years I think is the max).  Therefore, most software companies tend to update the copyright to the most recent date or date when the software will be released (e.g. if it's next year vs. this year).

Ideally, instead of hardcoding this, another way to do this is to have a file with all copyright dates for a given project so that the various components (entire project) can reference this "copyright" file with the correct copyright date wherever applicable.  Thoughts?

Reply:
Thanks @julienlim! If I understood you well, that file would be more or less the same as hardcoding it, wouldn't it? @a2batic implemented a build-timestamped copyright. Is that ok?